### PR TITLE
Implement TeiDocument validation with tests (header/body clashes and cast)

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- Adds validation for TeiDocument in tei-core, including internal checks and test coverage
- Enforces cross-checks between header annotation systems and body identifiers to catch duplicates across annotation systems and body blocks
- Validates utterance speakers against the profile cast with nuanced handling for empty vs. absent casts
- Introduces behaviour-driven tests and helper utilities to exercise new validation scenarios

### Core
- Implemented TeiDocument::validate() in tei-core to perform internal validation checks such as:
  - ensuring unique xml:id values across header annotation systems and body blocks
  - detecting clashes between header annotation systems and body blocks
  - validating utterance speakers against the profile cast when a cast is declared
  - treating an empty cast as declared (requiring speakers to be present) while the absence of a cast allows speaker references for incremental validation
- Validation results surface as TeiError::Validation where applicable

### Tests
- Added new validation scenarios in tei-core/tests/features/validation.feature:
  - Rejecting header and body identifier clashes
  - Rejecting speakers when the cast is empty
  - Allowing speakers when the cast list is absent
- Extended tei-core/tests/validation_behaviour.rs with test helpers and fixtures:
  - the_profile_is_declared_but_empty
  - the_encoding_includes_annotation_system
  - declare_empty_profile, add_annotation_system helpers
- Introduced test utilities to manipulate header encoding and profile state to exercise new validation paths

### Documentation
- Updated docs/roadmap.md to reflect that TeiDocument::validate() is implemented
- Updated docs/tei-rapporteur-design-document.md with notes about new validation behaviours and scenarios
- Updated docs/users-guide.md to describe the new validate() semantics, including empty vs absent cast edge cases and cross-identifier checks

## Test plan
- Run cargo test -p tei-core
- Verify tests pass for:
  - unique and duplicate xml:id checks across header and body
  - header/body id clash rejection
  - rejection when the cast is empty (invalid speaker references)
  - allowance when the cast is absent (speakers may be referenced for drafts)
- Validate that validation feature scenarios surface correct error messages, e.g. "duplicate xml:id 'shared'" and "utterance speaker 'ghost' is not declared in the profile".

If you want, I can run through any particular test scenario locally or adjust messages to align with project conventions.

🌿 Generated by [Terry](https://www.terragonlabs.com)

📎 **Task**: https://www.terragonlabs.com/task/aecc31a8-7e2f-4196-b335-d96b29a9a1ef

## Summary by Sourcery

Add TEI document validation coverage and document the new validation semantics across header/body identifiers and profile cast handling.

New Features:
- Add behaviour-driven validation scenarios for TEI documents covering header/body identifier clashes and profile cast semantics.

Enhancements:
- Extend test helpers to configure empty profiles and header annotation systems for validation scenarios.
- Update documentation to describe validate() behaviour, including header/body identifier checks and empty vs. absent cast edge cases.
- Mark TeiDocument::validate() as implemented in the project roadmap.

Documentation:
- Document TeiDocument::validate() semantics in the user guide and design document, including identifier and cast validation rules.

Tests:
- Expand validation.feature and validation_behaviour.rs with new BDD scenarios and helpers for identifier and cast validation behaviour.

### CODE REVIEW

File: tei-core/tests/validation_behaviour.rs [hunk 1]
```diff
@@ -102,6 +102,27 @@ fn the_profile_includes_speaker(
...
+#[expect(
+    clippy::needless_pass_by_value,
+    reason = "rstest_bdd supplies owned Strings for captured step parameters."
+)]
+fn the_encoding_includes_annotation_system(
+    #[from(validated_state)] state: &ValidationState,
+    identifier: String,
+) -> Result<()> {
+    state.update_document(|document| {
+        add_annotation_system(document, identifier.as_str(), "annotations")
```
[the remark this answers is not in the document]
Comment (leynos) [reply tei-core/tests/validation_behaviour.rs]: Has this comment now been resolved?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API.

File: tei-core/tests/validation_behaviour.rs [hunk 2]
```diff
@@ -4,7 +4,7 @@ use anyhow::{Context, Result, anyhow, ensure};
 use rstest::fixture;
 use rstest_bdd_macros::{given, scenario, then, when};
 use std::cell::RefCell;
-use tei_core::{P, ProfileDesc, TeiDocument, TeiError, Utterance};
+use tei_core::{AnnotationSystem, EncodingDesc, P, ProfileDesc, TeiDocument, TeiError, Utterance};
```
[the remark this answers is not in the document]
Comment (leynos) [reply tei-core/tests/validation_behaviour.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API.

File: docs/users-guide.md [hunk 3]
```diff
@@ -16,8 +16,12 @@ available today and how to exercise it.
...
   `validate()` to enforce document-wide rules: it rejects duplicate `xml:id`
   values across annotation systems, paragraphs, and utterances, and ensures
-  utterance speakers appear in the profile cast when it exists. Violations
-  surface as `TeiError::Validation`.
+  utterance speakers appear in the profile cast when it exists. An empty cast
+  still counts as declared—every `who` fails until the speakers are populated—
+  whereas the absence of a cast allows speaker references so drafts can be
+  validated incrementally. Identifier checks span the header as well, catching
+  clashes between annotation systems and body blocks. Violations surface as
+  `TeiError::Validation`.
```
[the remark this answers is not in the document]
Comment (leynos) [reply docs/users-guide.md]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API.
